### PR TITLE
Add transaction isolation level support

### PR DESCRIPTION
**Pull request by Arkatufus**
## Changes

* Add `IsolationLevel` properties to SqlJournalOptions and SqlSnapshotOptions base class
* Add `IsolationLevel` properties to Akka.Persistence.SqlServer.Hosting
* Add `IsolationLevel` properties to Akka.Persistence.PostgreSql.Hosting

### CODE REVIEW

File: src/Akka.Hosting.API.Tests/verify/CoreApiSpec.ApprovePersistence.verified.txt [hunk 1]
```diff
@@ -64,19 +69,23 @@ namespace Akka.Persistence.Hosting
         public string ConnectionString { get; set; }
         public System.TimeSpan ConnectionTimeout { get; set; }
         public abstract string MetadataTableName { get; set; }
+        public abstract System.Data.IsolationLevel ReadIsolationLevel { get; set; }
```
Comment (Aaronontheweb) [inline src/Akka.Hosting.API.Tests/verify/CoreApiSpec.ApprovePersistence.verified.txt:72]: Shouldn't this only be in the Sql-specific plugins?

**Review by Aaronontheweb — COMMENTED** [review]
Left a question

File: src/Akka.Hosting.API.Tests/verify/CoreApiSpec.ApprovePersistence.verified.txt (hunk 1 above)
Comment (Arkatufus) [reply src/Akka.Hosting.API.Tests/verify/CoreApiSpec.ApprovePersistence.verified.txt:72]: Yes, this change is inside `SqlJournalOptions` class

Comment (Aaronontheweb) [reply src/Akka.Hosting.API.Tests/verify/CoreApiSpec.ApprovePersistence.verified.txt:72]: Ah - I didn't realize those types were in the base Akka.Persistence.Hosting package.

**Pull request merged**